### DISCUSSION TITLE
[Reading Group]: RLP: Reinforcement as a Pretraining Objective

### DISCUSSION THREAD

**Discussion by ricj** [dsai-nlp/core#34] [Meetings & Announcements]
### 📆 Date & Time

Monday, December 8, 15-17

### 📍 Location or Link

EDIT-8103 and Zoom (https://chalmers.zoom.us/j/67926488196, passcode 314171)

### 📄 Paper Link

https://arxiv.org/pdf/2510.01265

### 📖 Curator

Nicklas?

**Comment by ricj** [comment]
Page 3: why is the no-think baseline implemented as an exponential moving average (EMA)?

**Reply by ricj** [reply]
Found it: it is explained at the bottom of page 3.

**Comment by ricj** [comment]
They claim that RLP gives a "dense reward". But is that actually true? The reward is given for every next token prediction, and they don't give a reward for every token in the CoT.

**Comment by NicklasOsterbacka** [comment]
Pretty cool paper, with some clear advantages in benchmarks!

1. During training, the model learns to reason about the next token to my understanding. Is that also done at inference time? I don't see how that necessarily induces more complex reasoning that may be required for some tasks. 
2. Related to that, how are the benchmarks structured? Is a single-token completion sufficient to solve most of them? They show the CoT for one example on the very last page of the ICLR manuscript, but that seems to be from a pre-training example.
3. Is CoT for every next-token prediction optimal? To me, some sort of sentence-based variant sounds more in line with "traditional" LLM reasoning.
4. Do their baselines reflect how reasoning LLMs are conventionally trained? If not, would RLP offer any performance advantage over such models?

**Comment by nicaudinet** [comment]
I liked it! Some comments and questions:

- How expensive is it to pre-train this way? Of course NVidia would sponsor GPU-hungry methods ...
- At what point in pre-training does the CoT actually start being useful?
- In Table 1, interesting that the scores of some datasets actually go _down_ after more continuous pre-training
- The EMA teacher seems like the magic dust of this paper
